src/waveform/renderers/allshader/waveformrendererslipmode.h

```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+#include <QColor>
+#include <memory>
+
+#include "shaders/slipmodeshader.h"
+#include "util/class.h"
+#include "util/performancetimer.h"
+#include "waveform/renderers/allshader/waveformrenderer.h"
+
+class ControlProxy;
+class QDomNode;
+class SkinContext;
+
+namespace allshader {
+class WaveformRendererSlipMode;
```

Comment:
Is this forward declaration redundant?

Reply:
It looks like it is - the class cannot be declared otherwise. I tried declaring in the namespace block but then the base class cannot resolve.